rest-service/src/main/kotlin/mb/api/service/valid/SequenceTransformer.kt

```diff
@@ -0,0 +1,111 @@
+package mb.api.service.valid
+
+import java.util.Scanner
+
+/**
+ * Max line length for a sequence line.
+ */
+private const val SequenceLineLength = 80
+
+object Sequences {
+
+  /**
+   * Standardizes the input query into a query consisting of only the query
+   * sequence.
+   *
+   * The output will have no blank lines, no extra whitespaces, and will be
+   * broken into 80 character rows.
+   *
+   * Header lines will not be altered.
+   *
+   * Input:
+   * ```
+   * > Some really long header line containing a bunch of information that is more than 80 characters long.
+   * aaaaaaaaaaaaaa
+   * aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+   *
+   * aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+   * a
+   * ```
+   *
+   * Output
+   * ```
+   * > Some really long header line containing a bunch of information that is more than 80 characters long.
+   * aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+   * aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+   * aaaaaaaaaaaaaaaaaaaaa
+   * ```
+   */
+  fun standardize(input: String): String {
+
+    // Input sequence scanner.
+    val scanner = Scanner(input)
+
+    // Buffer used to standardize lines to a set number of characters.
+    val lineBuffer = StringBuilder(SequenceLineLength)
+
+    // Output writer.
+    val writer = StringBuilder(input.length)
+
+    // Segment of an input line that could not fit into the line buffer.
+    var overflow: String
+
+    while (scanner.hasNextLine()) {
+      // Current line
+      val line     = scanner.nextLine().trim()
+
+      // If the line starts with a '>' character then it is a FASTA header line
+      // and should be written to the output stream unchanged.
+      if (line.startsWith('>')) {
+        writer.appendLine(line)
+
+        // If the line is empty, skip it, else write it out.
+      } else if (line.isNotBlank()) {
+        overflow = appendBuffer(lineBuffer, line)
+
+        // If the overflow string is not empty, then the line buffer must be
+        // full.
+        while (overflow != "") {
```

Comment:
Not sure what the purpose of overflow is here; it's defined outside the loop so I thought it would contain partial lines not yet written to the writer (which may be further appended to up to max_length).  But you don't get out of the inner loop without overflow being empty.  I don't see how you're tracking the length of the partial line written to the writer so you know how many more chars to write before the next newline.